server/netty-server/src/main/scala/sttp/tapir/server/netty/internal/NettyToResponseBody.scala

```diff
@@ -1,49 +1,80 @@
 package sttp.tapir.server.netty.internal
 
-import io.netty.buffer.{ByteBuf, Unpooled}
+import io.netty.buffer.Unpooled
+import io.netty.handler.codec.http.HttpChunkedInput
+import io.netty.handler.stream.ChunkedFile
 import sttp.capabilities
 import sttp.model.HasHeaders
 import sttp.tapir.capabilities.NoStreams
 import sttp.tapir.server.interpreter.ToResponseBody
-import sttp.tapir.{CodecFormat, RawBodyType, WebSocketBodyOutput}
+import sttp.tapir.server.netty.NettyResponse
+import sttp.tapir.{CodecFormat, FileRange, RawBodyType, WebSocketBodyOutput}
 
-import java.io.InputStream
+import java.io.{InputStream, RandomAccessFile}
 import java.nio.ByteBuffer
 import java.nio.charset.Charset
-import java.nio.file.Files
 
-class NettyToResponseBody extends ToResponseBody[ByteBuf, NoStreams] {
+class NettyToResponseBody extends ToResponseBody[NettyResponse, NoStreams] {
   override val streams: capabilities.Streams[NoStreams] = NoStreams
 
-  override def fromRawValue[R](v: R, headers: HasHeaders, format: CodecFormat, bodyType: RawBodyType[R]): ByteBuf = {
+  override def fromRawValue[R](v: R, headers: HasHeaders, format: CodecFormat, bodyType: RawBodyType[R]): NettyResponse = {
     bodyType match {
       case RawBodyType.StringBody(charset) =>
-        Unpooled.wrappedBuffer(v.toString.getBytes(charset))
+        val bytes = v.asInstanceOf[String].getBytes(charset)
+        Left(Unpooled.wrappedBuffer(bytes))
+
       case RawBodyType.ByteArrayBody =>
         val bytes = v.asInstanceOf[Array[Byte]]
-        Unpooled.wrappedBuffer(bytes)
+        Left(Unpooled.wrappedBuffer(bytes))
+
       case RawBodyType.ByteBufferBody =>
         val byteBuffer = v.asInstanceOf[ByteBuffer]
-        Unpooled.wrappedBuffer(byteBuffer)
+        Left(Unpooled.wrappedBuffer(byteBuffer.array()))
 
       case RawBodyType.InputStreamBody =>
         val stream = v.asInstanceOf[InputStream]
-        Unpooled.wrappedBuffer(stream.readAllBytes())
+        Left(Unpooled.wrappedBuffer(stream.readAllBytes()))
```

Comment:
here we are still reading everything into memory - maybe we need a third case in `NettyResponse` :)